server/router/middleware/session/org.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package session
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+
+	"go.woodpecker-ci.org/woodpecker/v2/server/model"
+	"go.woodpecker-ci.org/woodpecker/v2/server/store"
+	"go.woodpecker-ci.org/woodpecker/v2/server/store/types"
+)
+
+func Org(c *gin.Context) *model.Org {
+	v, ok := c.Get("org")
```

Comment:
This one only works now if `MustOrg` was called previously.

Reply:
yes as with other middleware ...

Reply:
it's  a feature not a bug ... I understand that it could be frigned if you just read the functions behind endpints ... but we do the dame to ensure permissions are set and respected etc ...

... this decuples the endpoint function a bit and also allow tests to be more easy to write agains it ...

Reply:
> but we do the dame to ensure permissions are set and respected etc ...

Actually not, we always call `SetUser`, `SetPerm`, `SetRepo` in the router to add sth to the context and later on, load it using `MustUser` or `User`. Can I push the change to adjust it this way?

Reply:
i dont get it :/ so ... Sure push it ...